### PR TITLE
Added log messages to reflect deactivation procedure

### PR DESCRIPTION
**Pull request by NanoAkron**
Better step-wise logging for future debugging

### CODE REVIEW

**Comment by moneromooo-monero** [comment]
ok, though I'd have added the logs in the relevant functions, rather than after the caller, so that they always trigger when the action is performed (and some already have such logs, like miner stop).

**Comment by NanoAkron** [comment]
Makes more sense. Will fix tomorrow.

**Comment by moneromooo-monero** [comment]
If I chose the shed color, I'd put 1 for the important ones, but seems ok too.

**Comment by NanoAkron** [comment]
So this now does the logging inside the functions that are called at shutdown.

Additional change to line 390 of `src/cryptonote_core/blockchain.cpp` for consistency with all other functions in there because it was sat right in front of me. I can go through and change them all to `yellow` if preferred.

Interesting to note re: `core::deinit()` within `src/cryptonote_core/cryptonote_core.cpp`

`m_mempool.deinit()` seemingly does nothing - only the first log entry (`"Received signal to deactivate memory pool store"`) is made, none of the others ever appear.

`m_protocol.deinit()` does nothing - I can only trace it back to an empty template (`t_cryptonote_protocol_handler`) in `src/cryptonote_protocol/cryptonote_protocol_handler.inl` - let me know if I'm missing something here - hence why I put the log message in the destructor

File: src/cryptonote_protocol/cryptonote_protocol_handler.inl [hunk 1]
```diff
@@ -75,8 +75,6 @@ namespace cryptonote
   template<class t_core>
   bool t_cryptonote_protocol_handler<t_core>::deinit()
   {
-
```
Comment (NanoAkron) [inline src/cryptonote_protocol/cryptonote_protocol_handler.inl]: Must have done this whitespace removal when fiddling - not on purpose

**Pull request merged**